### PR TITLE
Button component variants from YAML file

### PR DESCRIPTION
**Pull request by igloosi**
This PR:
- adds a YAML file where default option and variants are defined
- updates macro, template and documentation to read and display those variants

### CODE REVIEW

File: src/components/button/button.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    classes: 'boom'
+- name: disabled
+  data:
+    text: Disabled button
+    isDisabled: true
+- name: start
+  data:
+    text: Start now
+    url: /
+    isStart: true
```
Comment (fofr) [inline src/components/button/button.yaml]: Is `is` needed in `isStart` and `isDisabled`?
On GOV.UK we tried to mirror the DOM where possible, "disabled" would follow that approach.

File: src/components/button/button.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+variants:
```
Comment (fofr) [inline src/components/button/button.yaml]: Are variants the same as examples?

After research with GOV.UK developers we changed our equivalent from "fixtures" to "examples". (https://github.com/alphagov/govuk_publishing_components/pull/37)

File: src/components/button/button.yaml (hunk 1 above)
Comment (igloosi) [reply src/components/button/button.yaml]: I can do either way with this one. Seeing we're proposing is- prefix for component states I'd like to follow similar convention. We will be testing this on devs.

File: src/components/button/button.yaml (hunk 2 above)
Comment (igloosi) [reply src/components/button/button.yaml]: Interesting. was fixtures more obvious than variants? 
yes, variants also referrers to examples

Comment (fofr) [reply src/components/button/button.yaml]: We didn't try variants. People referred to them as examples, so we used the term they chose.

There may be some subtle differences with how you're using the YAML files. I think it'd be useful to keep YAML structures the same to keep components interoperable.

**Pull request merged**